be/src/exec/hdfs_scanner.cpp

```diff
@@ -453,13 +440,9 @@ void HdfsScannerContext::append_or_update_partition_column_to_chunk(ChunkPtr* ch
             }
             chunk_part_column->assign(row_count, 0);
         }
-
-        if (ck->is_slot_exist(slot_desc->id())) {
-            ck->update_column(std::move(chunk_part_column), slot_desc->id());
-        } else {
-            ck->append_column(std::move(chunk_part_column), slot_desc->id());
-        }
+        ck->append_or_update_column(std::move(chunk_part_column), slot_desc->id());
     }
+    ck->set_num_rows(row_count);
```

Comment:
do we need this?

Reply:
not necessarily, but I guess there is no harm to do that.